kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/RowTracking.java

```diff
@@ -187,14 +190,32 @@ public static List<DomainMetadata> updateRowIdHighWatermarkIfNeeded(
             AddFile addFile = new AddFile(row.getStruct(SingleAction.ADD_FILE_ORDINAL));
             if (!addFile.getBaseRowId().isPresent()
                 || addFile.getBaseRowId().get() > prevRowIdHighWatermark) {
-              currRowIdHighWatermark.addAndGet(getNumRecordsOrThrow(addFile));
+              currCalculatedRowIdHighWatermark.addAndGet(getNumRecordsOrThrow(addFile));
             }
           }
         });
 
-    if (currRowIdHighWatermark.get() != prevRowIdHighWatermark) {
+    // If the txn builder has explicitly provided a row ID high watermark, we should use that value
```

Comment:
Why do we need to do a conflict resolution when the user provides a high watermark? I think the intention when they provide a high watermark is the connector wants to explicitly dictate the high watermark and the commit contents. Rebasing when conflicts could create unintended problems and can make the code too complex.